spec/spec_helper.rb

```diff
@@ -1,3 +1,33 @@
 # encoding: utf-8
 require "logstash/devutils/rspec/spec_helper"
 require 'logstash/inputs/unix'
+
+class UnixSocketHelper
+
+  attr_reader :path
+
+  def initialize
+    @socket = nil
+  end
+
+  def new_socket(path)
+    @path   = path
+    File.unlink if File.exists?(path) && File.socket?(path)
+    @socket = UNIXServer.new(path)
+    self
+  end
+
+  def loop(forever=false)
+    @thread = Thread.new do
+      s = @socket.accept
+      s.puts "hi" while forever
+    end
+    self
+  end
+
+  def close
+    @thread.kill
+    @socket.close
+    File.unlink(path)
+  end
+end
```

Comment:
I think they are in a good place now, spec helpers are good place to put this kind of code, there is only benefit of doing what you say when you have several of them.